### PR TITLE
Add a `allow_asm` option so virtual ISA based targets (JS/PNaCl/WAsm) can disallow the asm! macro.

### PR DESCRIPTION
**Pull request by DiamondLovesYou**

### CODE REVIEW

File: src/librustc_back/target/mod.rs [hunk 1]
```diff
@@ -168,6 +168,8 @@ pub struct TargetOptions {
     /// currently only "gnu" is used to fall into LLVM. Unknown strings cause
     /// the system linker to be used.
     pub archive_format: String,
+    /// Is asm!() not allowed? Defaults to false
+    pub no_asm: bool,
```
Comment (alexcrichton) [inline src/librustc_back/target/mod.rs]: Could this instead be inverted to `allow_asm`? Reading a double negative is sometimes a little confusing. Other than that though looks good to me!

Comment (DiamondLovesYou) [inline src/librustc_back/target/mod.rs]: Fair enough. Done.

**Comment by kryptan** [comment]
Why would `asm!` need to be dissalowed on virtual ISA based targets? Shouldn't it just allow you to write in traget ISA?

**Pull request merged**